### ISSUE TITLE
Question about "A configuration has no node"

### ISSUE THREAD

**Issue by DrawZeroPoint** [humanoid-path-planner/hpp-manipulation#119]
Dear HPP developers,

When using the `solve` function of `ProblemSolver`, I occasionally come across the "A configuration has no node" error. After searching through the repositories, I assume the error is raised here:

https://github.com/humanoid-path-planner/hpp-manipulation/blob/103d495dfeb97fbee50073ed9a46bc4a9f51e877/src/graph/state-selector.cc#L88

So, does this error means the configurations I set to the solver using `setInitialConfig` (since the error msg points to the initial configs) are somehow wrong? Regarding the values, I saw no obvious fault like collision or joint limit violation. I assume there is a deeper reason for this. If so, how could we detect which part of those is wrong and correct them?

I tried to use the `applyNodeConstraints` to hopefully convert the unusable configurations into valid ones, but the output is pretty close to the original values where `res=True` and the `error` is approximately 1e-7. As a result, the output can still cause the error.

I have tried to use `clearConfigValidations` for the solver when doing the tests, but that did not avoid the error.

Gratefully thank you for any insights and corrections about the problem and what I misunderstood. I am willing to provide more details if necessary.

**Comment by florent-lamiraux** [comment]
The error message indeed means that one configuration in the roadmap does not satisfy the constraints of any node of the constraint graph. To check this beforehand, you can call `graph.getNode(q)` to know in which node (or state) configuration `q` lies if any.
I am surprised however that calling `applyNodeConstraints` you do not get a configuration that lies in the given node if the function returns success.
Can you type the following commands, check that `res` is `True`  and return the result ?
```
node =
res, q, err = graph.applyNodeConstraints(node, q_init)
graph.getNode(q)
```
where `node` is the node where you configuration should lie in ?

**Comment by DrawZeroPoint** [comment]
Dear @florent-lamiraux,

Thank you for the instructions that let me out! To summarize, it turned out to be the joint values of the gripper violated the predefined `lock hand` constraint. In my code, these values should be 0.04 given the constraint, but since I used measured (from simulation) values to fill them, that could drift to ~0.36 and break the constraint at runtime given the error threshold being 1e-3. 

Such a violation does not relevant to joint limits (the Franka gripper's range is [0, 0,04]), so it is reasonable that config validation cannot prevent that to step into the solving process. 

The lesson I learned from this is always to be careful with the `lock hand` constraint because in my first proposal I failed to catch the error. The projected configuration, as you said, does lie inside the `free` node, and the only correction happened to the gripper joint states, which eventually let me discover the cause.
